sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5663,9 +5767,27 @@ pi_result piextUSMHostAlloc(void **ResultPtr, pi_context Context,
   return PI_SUCCESS;
 }
 
-pi_result piextUSMFree(pi_context Context, void *Ptr) {
+static pi_result piextUSMFreeImpl(pi_context Context, void *Ptr) {
+  pi_platform Plt = Context->Devices[0]->Platform;
+  auto MemAlloc =
+      std::find_if(std::begin(Plt->MemAllocs), std::end(Plt->MemAllocs),
+                   [&](MemAllocRecord &Rec) { return Rec.Ptr == Ptr; });
+  if (MemAlloc == std::end(Plt->MemAllocs)) {
+    die("All memory allocations must be tracked!");
+  }
+  if (--(MemAlloc->RefCount) != 0) {
```

Comment:
why is this refCount not protected by lock? it could be that there's a thread trying to free it and another thread executing a kernel with it, right?